### PR TITLE
feat: [CI-4620]: Added support for git status push for azure repo 

### PR DESCRIPTION
**Pull request by raghavharness**
## Describe your changes
Added support for git status push for azure repo 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34164)

### CODE REVIEW

File: 320-ci-execution/src/main/java/io/harness/stateutils/buildstate/ConnectorUtils.java [hunk 1]
```diff
@@ -689,6 +694,20 @@ private String fetchUserNameFromGitlabConnector(GitlabConnectorDTO gitConfigDTO)
     return null;
   }
 
+  private String fetchUserNameFromAzureRepoConnector(AzureRepoConnectorDTO gitConfigDTO) {
```
Comment (harshjain12) [inline 320-ci-execution/src/main/java/io/harness/stateutils/buildstate/ConnectorUtils.java:697]: Handle usernameRef

File: 320-ci-execution/src/main/java/io/harness/stateutils/buildstate/ConnectorUtils.java [hunk 2]
```diff
@@ -689,6 +694,20 @@ private String fetchUserNameFromGitlabConnector(GitlabConnectorDTO gitConfigDTO)
...
+    if (gitConfigDTO.getAuthentication().getAuthType() == GitAuthType.HTTP) {
```
Comment (harshjain12) [inline 320-ci-execution/src/main/java/io/harness/stateutils/buildstate/ConnectorUtils.java:698]: send null

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/ci/CIBuildStatusPushTask.java [hunk 3]
```diff
@@ -70,7 +80,10 @@ public class CIBuildStatusPushTask extends AbstractDelegateRunnableTask {
   private static final String GITHUB_API_URL = "https://api.github.com/";
   private static final String BITBUCKET_API_URL = "https://api.bitbucket.org/";
   private static final String GITLAB_API_URL = "https://gitlab.com/api/";
+  private static final String AZURE_REPO_API_URL = "https://dev.azure.com/";
+  private static final String AZURE_REPO_GENRE = "continuous-integration";
```
Comment (harshjain12) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/ci/CIBuildStatusPushTask.java:84]: Put HarnessCIE

File: 960-api-services/src/main/java/io/harness/git/GitClientHelper.java [hunk 4]
```diff
@@ -83,12 +83,12 @@
 @Slf4j
 public class GitClientHelper {
   private static final String GIT_URL_REGEX =
-      "(http|https|git|ssh)(:\\/\\/|@)([^\\/:]+(:\\d+)?)[\\/:]([^\\/:]+)\\/(.+)?(.git)?";
+      "(http|https|git|ssh)(:\\/\\/|@)([^\\/:]+(:\\d+)?)[\\/:](v\\d\\/)?([^\\/:]+)\\/(.+)?(.git)?";
```
Comment (raghavharness) [inline 960-api-services/src/main/java/io/harness/git/GitClientHelper.java:86]: @DeepakPatankar Can you please help in review of this change.

Comment (DeepakPatankar) [reply 960-api-services/src/main/java/io/harness/git/GitClientHelper.java:86]: Can you add UTs to ensure that this regex works for all the type of URLs.

1. Github SaaS
2. Github on-prem
3. Bitbucket SaaS
4. Bitbucket Server
5. Azure Repo

Comment (raghavharness) [reply 960-api-services/src/main/java/io/harness/git/GitClientHelper.java:86]: SSH url for is like **git@ssh.dev.azure.com:v3/org/project/repo**

Comment (raghavharness) [reply 960-api-services/src/main/java/io/harness/git/GitClientHelper.java:86]: Sure I'll add it.

Comment (raghavharness) [reply 960-api-services/src/main/java/io/harness/git/GitClientHelper.java:86]: @DeepakPatankar Added

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/ci/CIBuildStatusPushTask.java (hunk 3 above)
Comment (raghavharness) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/ci/CIBuildStatusPushTask.java]: done

**Review by DeepakPatankar — DISMISSED** [review]

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 11 of 15 files at r1, 4 of 6 files at r3, 1 of 3 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34164#-:-N4VjhGF8w2RFDeHMU6y:bnfp4nl)* status: 2. unresolved discussions.

**Review by harshjain12 — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34164#-:-N4Vjn9M3UdKUS89HTtW:bnfp4nl)* status: 1. unresolved discussions.

**Review by harshjain12 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34164)* status: 1. unresolved discussions.

**Review by raghuAtWings — APPROVED** [review]

**Review by gaurav-jain1 — APPROVED** [review]
> Reviewed 11 of 15 files at r1, 3 of 6 files at r3, 1 of 3 files at r4, 3 of 3 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34164)* status: 1. unresolved discussions.

**Pull request merged**